web/app/js/components/ResourceDetail.jsx

```diff
@@ -149,7 +149,9 @@ export class ResourceDetailBase extends React.Component {
         // downstream resources of this resource (meshed traffic only)
         this.api.fetchMetrics(
           `${this.api.urlsForResource("all")}&from_name=${resource.name}&from_type=${resource.type}&from_namespace=${resource.namespace}`
-        )
+        ),
+        // definition for this resource
+        this.api.fetchResourceDefinition(resource.namespace, resource.type, resource.name),
```

Comment:
Hi @scottcarol,

I’ve chained the new resource definition api call to the other ones in that component, so that all of them are called the same way. But the response of this new one is a bit large and maybe we don’t need to call it that often. What do you think?

Cheers

Reply:
Great question @cynthia-sg! Having this code in the `ResourceDetail` component is great for testing. Because the goal of this project is to prepare for future work but not to display any data, I agree with you that we shouldn't make unnecessary queries. 

What about adding `queryForDefinition: false` to the state, and check for `true` before adding `fetchResourceDefinition` to the `apiRequests` array? That way your code is ready to run once we're utilizing the data. Open to other suggestions as well!